CMSgov/cms-carts-seds
# Enable field 2 in Basic State Information

### Description
Make Question 2, Program Type, in Basic State Information editable, and update corresponding logic to make form react to this update

Three tables hold information for `programType`:
- state
- sections
- state-status


state – holds a static value and thus we should seek to neither update nor read `programType` from this in the app's flow
sections – no change, holds the `programType` value from the user's choice in the UI
state-status – is now updated with the value from sections, and is the sole reference for that value in the frontend because it is accessible to all user types

### Related ticket(s)
MDCT-2993

---
### How to test

- Login as admin
- Generate 2022 and 2023 forms
- Login as stateuser2 and select the 2023 form
- Question two of Basic State Information should be editable
- The blue banners in Section 1 should react to the changes (Part 1 and 3 get bannered when you select the third option; parts 2 and 4 when you choose the second; none bannered for the first). There are other pages in the form that also change as a result and they can be identified by searching `show_if_state_program_type_in` in seed-section-base-2023.json

---
### Author checklist


---
